### PR TITLE
Legacy checkouts

### PR DESCRIPTION
**Pull request by dholms**

### CODE REVIEW

**Review by devinivy — APPROVED** [review]

File: packages/pds/src/db/index.ts [hunk 1]
```diff
@@ -193,8 +193,17 @@ export class Database {
...
     this.assertTransaction()
+    return this.txAdvisorylock(name)
+  }
+
+  async checkTxAdvisoryLock(name: string): Promise<boolean> {
+    this.assertNotTransaction()
+    return this.txAdvisorylock(name)
+  }
+
+  private async txAdvisorylock(name: string): Promise<boolean> {
```
Comment (devinivy) [inline packages/pds/src/db/index.ts:206]: Just a tiny consistency thing.
```suggestion
    return this.txAdvisoryLock(name)
  }

  async checkTxAdvisoryLock(name: string): Promise<boolean> {
    this.assertNotTransaction()
    return this.txAdvisoryLock(name)
  }

  private async txAdvisoryLock(name: string): Promise<boolean> {
```

Comment (dholms) [reply packages/pds/src/db/index.ts]: yup good catch

File: packages/pds/src/db/index.ts [hunk 2]
```diff
@@ -193,8 +193,17 @@ export class Database {
...
 
-  async txAdvisoryLock(name: string): Promise<boolean> {
+  async takeTxAdvisoryLock(name: string): Promise<boolean> {
     this.assertTransaction()
+    return this.txAdvisorylock(name)
+  }
+
+  async checkTxAdvisoryLock(name: string): Promise<boolean> {
+    this.assertNotTransaction()
+    return this.txAdvisorylock(name)
```
Comment (devinivy) [inline packages/pds/src/db/index.ts]: Clever/interesting—I assume it takes and releases the lock during the course of the statement?  I wonder if there's a simple way to check without taking it at all. 🤔 May not really matter!

File: packages/pds/src/services/repo/index.ts [hunk 3]
```diff
@@ -159,6 +164,12 @@ export class RepoService {
     writes: PreparedWrite[],
     swapCommit?: CID,
   ): Promise<CommitData> {
+    // this is not in a txn, so this won't actually hold the lock,
+    // we just check if it is currently held by another txn
+    const available = await storage.lockAvailable()
+    if (!available) {
+      throw new ConcurrentWriteError()
+    }
```
Comment (devinivy) [inline packages/pds/src/services/repo/index.ts:172]: I take it this is defensive, i.e. not really required but allows us to fail early?

Comment (dholms) [reply packages/pds/src/services/repo/index.ts:172]: yup exactly

**Review by devinivy — APPROVED** [review]

**Pull request merged**